src/backends/tensorrt/plan_backend.cc

```diff
@@ -2693,9 +2717,11 @@ PlanBackend::Context::Run(
     }
   }
 
-  // Ensure inputs are ready before execution. Output buffers will always be
-  // available at this point as the execution and output copy are on the same
-  // stream.
+  // Ensure inputs are ready before execution.
+  // Output buffers are guaranteed to be available at this point when the
+  // execution and output copy are on the same stream. If a separate stream is
```

Comment:
"If a separate stream..."
I don't understand this sentence. Who is "the user"?

Reply:
The sentence is trying to say that if the model is configured to have a separate stream to do the output copy, then triton will assume that the output buffers can be overwritten by the inference. i.e there will be no explicit waiting on the output buffers being ready for overwrite.

Reply:
But triton is the controlling the whole thing so it is confusing as written. Triton is explicitly making sure that the output is available before inference is launched, either because the same stream is used, or because the sync you added... correct?

Reply:
The sync I added ensures that the output is available in the output device buffer to copy into the host buffer i.e the output copy stream waits on the output to be generated before it copies it into the host. However, there is no way currently for triton to ensure that the copy of the output from the output buffer to the host is complete before it starts inference.

Reply:
Then how can this change be safe?

Reply:
I don't think there is a safe way to overlap current TRT execution with output copy from previous inference using only one output binding buffer, because in the worse case scenario, we have to view the execution to be always writing to the buffer and the copy to be always reading from the buffer. This change is a "hacky" workaround to improve the large size output performance without ensuring correctness.